internal/job/checkout_test.go

```diff
@@ -0,0 +1,126 @@
+package job
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/buildkite/agent/v3/internal/job/githttptest"
+	"github.com/buildkite/agent/v3/internal/shell"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDefaultCheckoutPhase(t *testing.T) {
+	assert := require.New(t)
+	ctx := context.Background()
+
+	shell, err := shell.New()
+	assert.NoError(err)
+
+	tests := []struct {
+		name        string
+		executor    *Executor
+		projectName string
+		checkoutDir string
+		refSpec     string
+		expectErr   bool
```

Comment:
I think this is currently unused?

Reply:
@catkins yeah I want to add some error cases, can remove for now.